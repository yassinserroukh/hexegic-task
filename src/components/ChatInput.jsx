import React, { useState } from 'react';

const ChatInput = ({ onSendMessage, isLoading, searchMode, onModeChange }) => {
  const [inputValue, setInputValue] = useState('');

  const handleSubmit = () => {
    if (inputValue.trim() && !isLoading) {
      onSendMessage(inputValue.trim());
      setInputValue('');
    }
  };

  return (
    <div className="input-container">
      <div className="input-wrapper">
        <input
          type="text"
          className="input-field"
          value={inputValue}
          onChange={(e) => setInputValue(e.target.value)}
          disabled={isLoading}
        />
        <button
          className="send-button"
          onClick={handleSubmit}
          disabled={!inputValue.trim() || isLoading}
        >
          {isLoading ? 'Sending...' : 'Send'}
        </button>
      </div>

      <div className="settings">
        <div>
          Mode:
          <select
            className="mode-selector"
            value={searchMode}
            onChange={(e) => onModeChange(e.target.value)}
            disabled={isLoading}
          >
            <option value="contextual">Contextual AI</option>
            <option value="semantic">Semantic Search</option>
            <option value="regular">Direct AI</option>
          </select>
        </div>
      </div>
    </div>
  );
};

export default ChatInput;
