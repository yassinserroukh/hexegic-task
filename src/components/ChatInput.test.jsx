import React from 'react';
import { render, screen } from '@testing-library/react';
import ChatInput from './ChatInput';
import userEvent from '@testing-library/user-event';

describe('ChatInput', () => {
  it('should render the input field and send button', () => {
    const { container } = render(
      <ChatInput
        onSendMessage={jest.fn()}
        isLoading={false}
        searchMode="contextual"
        onModeChange={jest.fn()}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should update the input value when typed into', async () => {
    render(
      <ChatInput
        onSendMessage={() => {}}
        isLoading={false}
        searchMode="contextual"
        onModeChange={() => {}}
      />,
    );

    const inputElement = screen.getByRole('textbox');
    await userEvent.type(inputElement, 'Hello world');
    expect(inputElement).toHaveValue('Hello world');
  });

  it('should call onSendMessage with the input value when send button is clicked', async () => {
    const mockOnSendMessage = jest.fn();
    render(
      <ChatInput
        onSendMessage={mockOnSendMessage}
        isLoading={false}
        searchMode="contextual"
        onModeChange={() => {}}
      />,
    );
    await userEvent.type(screen.getByRole('textbox'), 'Test message');
    await userEvent.click(screen.getByRole('button'));
    expect(mockOnSendMessage).toHaveBeenCalledWith('Test message');
  });

  it('should call onModeChange when the mode selector value changes', async () => {
    const mockOnModeChange = jest.fn();
    render(
      <ChatInput
        onSendMessage={() => {}}
        isLoading={false}
        searchMode="contextual"
        onModeChange={mockOnModeChange}
      />,
    );

    await userEvent.selectOptions(screen.getByRole('combobox'), 'semantic');

    expect(mockOnModeChange).toHaveBeenCalledTimes(1);
    expect(mockOnModeChange).toHaveBeenCalledWith('semantic');
  });
});
