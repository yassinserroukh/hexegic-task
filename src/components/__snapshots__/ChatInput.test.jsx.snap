// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChatInput should render the input field and send button 1`] = `
<div>
  <div
    class="input-container"
  >
    <div
      class="input-wrapper"
    >
      <input
        class="input-field"
        placeholder="Ask about UK cyber security, defence, or business documentation..."
        type="text"
        value=""
      />
      <button
        class="send-button"
        disabled=""
      >
        Send
      </button>
    </div>
    <div
      class="settings"
    >
      <div>
        Mode:
        <select
          class="mode-selector"
        >
          <option
            value="contextual"
          >
            Contextual AI
          </option>
          <option
            value="semantic"
          >
            Semantic Search
          </option>
          <option
            value="regular"
          >
            Direct AI
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
